datarepo-clienttests/src/main/java/runner/TestRunner.java

```diff
@@ -146,6 +146,23 @@ void executeTestConfigurationNoGuaranteedCleanup() throws Exception {
       throw new RuntimeException("Error calling test script setup methods.", setupExceptionThrown);
     }
 
+    // Failure Thread: fetch script specification if config is defined
+    if (config.failureScript != null) {
+      logger.debug("Creating thread pool for failure script.");
+      TestScript failureScript = config.failureScript.failureScriptClassInstance();
+      failureScript.setParameters(config.failureScript.parameters);
+
+      // create a thread pool for running its user journeys
+      ThreadPoolExecutor failureThreadPool = (ThreadPoolExecutor) Executors.newFixedThreadPool(1);
+      threadPools.add(failureThreadPool);
+
+      TestUserSpecification failureTestUser = config.testUsers.get(0);
+      UserJourneyThread ujt =
+          new UserJourneyThread(failureScript, config.failureScript.description, failureTestUser);
+      failureThreadPool.submit(ujt);
```

Comment:
I think the FailureScript setup/cleanup methods aren't getting called anywhere. RandomPodDelete doesn't override those methods, so I understand why it doesn't make a difference in this case. But I think it's confusing to allow failure scripts to extend that class if we don't call the methods in the same way. We could either:
- Call the setup/userJourney/cleanup methods similar to test scripts (i.e. setup before submit to pool, cleanup after pool shuts down and in the forcedcleanup wrapper method).
- Make a separate FailureScript abstract base class that just has a single method ("causeTrouble" or something) in place of the setup/userJourney/cleanup methods in TestScript. You'd probably still want some of the other methods from TestScript, too (setParameters, manipulatesKubernetes, setServer). One upside of this is that you could make the causeTrouble take the whole array of testusers instead of just picking the first one.

I lean towards the second option, but am not sure what the best choice is. What do you think?

Reply:
Great ideas.
-  I like the "cause trouble" idea. I think "failure script" is misleading, because things won't necessarily fail. But, I wanted something that is a single word. I propose "disrupt." Looking at its definition, I think it's rather fitting: "to interrupt (an event, activity, or process) by causing a disturbance or problem."
- I think the separate script abstract class is a good idea. 
- Looking at this, I found a bug with how I cleaning up the thread pools. I need to clean up the disruption thread pool and the test script thread pools, so I put them into a map and iterate through them to clean up.